### PR TITLE
Akeaveny/82/uwrt drive train ros controllers

### LINKED ISSUE #50: Add open loop control to drivetrain

### PR DESCRIPTION
**Pull request by akeaveny**
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
This branch contains a ROS Controller for the drive train

1. **Open Loop Position Controller**
- Subscribes to UWRTOpenLoopPosCmd Twist msg: linear & angular position commands
- Outputs: left & right position setpoints
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

TODO (akeaveny):
1. Open Loop Position Controller
- Look at scaling factor from Xbox Joystick
- try a ROS srv call to lock position offset from Xbox Joystick

closes #50

### CODE REVIEW

**Comment by wmmc88** [comment]
@akeaveny thanks for useful pr description. 

For #2, is there any reason u decided to recreate the diff drive controller? did u make any modifications to the package because we were already using the diff drive controller?

**Comment by akeaveny** [comment]
> @akeaveny thanks for useful pr description.
> 
> For #2, is there any reason u decided to recreate the diff drive controller? did u make any modifications to the package because we were already using the diff drive controller?

I deleted the custom diff drive controller as it was a duplicate of the standard ROS controller

File: uwrt_mars_rover_controllers/uwrt_open_loop_position_controller/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+build
```
Comment (wmmc88) [inline uwrt_mars_rover_controllers/uwrt_open_loop_position_controller/.gitignore:1]: what are these ignores for?

File: uwrt_mars_rover_msgs/msg/Vector3.msg [hunk 2]
```diff
@@ -0,0 +1,3 @@
+float32 x
```
Comment (wmmc88) [inline uwrt_mars_rover_msgs/msg/Vector3.msg:1]: Why are you redefining Vector3? its already defined in std:msgs

File: uwrt_mars_rover_hw/drivers/roboteq_driver/src/CanopenInterface.cpp [hunk 3]
```diff
@@ -306,7 +303,8 @@ DataType CanopenInterface::sendQuery(RuntimeQuery query, uint8_t subindex) {
     return response_data;
   }
 
-  std::cout << "QUERY RESPONSE ID" << response_frame.can_id << std::endl;
+  // std::cout << "QUERY RESPONSE ID" << response_frame.can_id << std::endl;
```
Comment (wmmc88) [inline uwrt_mars_rover_hw/drivers/roboteq_driver/src/CanopenInterface.cpp:306]: delete commented code

File: uwrt_mars_rover_hw/drivers/roboteq_driver/src/CanopenInterface.cpp [hunk 4]
```diff
@@ -254,36 +248,39 @@ DataType CanopenInterface::sendQuery(RuntimeQuery query, uint8_t subindex) {
...
+
+  if (static_cast<unsigned>(response_frame.can_dlc) != CAN_FRAME_SIZE_BYTES) {
+    throw std::runtime_error("Mismatched DLC value");
   }
 
   // TODO: throw appropriate errors
   if ((response_frame.data[0] & RESPONSE_TYPE_MASK) != SUCCESSFUL_QUERY_RESPONSE) {
-    std::cout << "Query unsuccessful response" << std::endl;
+    throw std::runtime_error("Unssuccessful query response");
   } else if (query_frame.data[1] != response_frame.data[1] || query_frame.data[2] != response_frame.data[2]) {
```
Comment (wmmc88) [inline uwrt_mars_rover_hw/drivers/roboteq_driver/src/CanopenInterface.cpp:280]: i believe one of your clang tidy failures is calling else after throws. This makes code harder to follow. 

It should be smth list:
```
if (error){
    throw error;
}

if (ohter_error){
    throw other_error;
}
```

and no need for else case cuz its like just the default if the block hasnt thrown an error

File: uwrt_mars_rover_controllers/uwrt_open_loop_position_controller/include/position_controllers/joint_group_position_controller.h [hunk 5]
```diff
@@ -0,0 +1,114 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2008, Willow Garage, Inc.
+ *  Copyright (c) 2012, hiDOF, Inc.
+ *  Copyright (c) 2013, PAL Robotics, S.L.
+ *  Copyright (c) 2014, Fraunhofer IPA
+ *  All rights reserved.
```
Comment (wmmc88) [inline uwrt_mars_rover_controllers/uwrt_open_loop_position_controller/include/position_controllers/joint_group_position_controller.h:8]: add ur name to copyright or add uw robotics and the current year. (as long as youve modified 3rd party code)

File: uwrt_mars_rover_controllers/uwrt_open_loop_position_controller/package.xml [hunk 6]
```diff
@@ -0,0 +1,29 @@
+<package format="2">
+  <name>uwrt_open_loop_position_controller</name>
+  <version>0.17.0</version>
+  <description>uwrt_open_loop_position_controller</description>
+
+  <maintainer email="bence.magyar.robotics@gmail.com">Bence Magyar</maintainer>
```
Comment (wmmc88) [inline uwrt_mars_rover_controllers/uwrt_open_loop_position_controller/package.xml:6]: delete the maintainers list and author. or just delete the maintainers and add urself as an author and maintainer

File: uwrt_mars_rover_controllers/uwrt_open_loop_position_controller/src/joint_group_position_controller.cpp [hunk 7]
```diff
@@ -0,0 +1,176 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2008, Willow Garage, Inc.
+ *  Copyright (c) 2012, hiDOF, Inc.
+ *  Copyright (c) 2013, PAL Robotics, S.L.
+ *  Copyright (c) 2014, Fraunhofer IPA
```
Comment (wmmc88) [inline uwrt_mars_rover_controllers/uwrt_open_loop_position_controller/src/joint_group_position_controller.cpp:7]: ditto ab copyright

**Comment by wmmc88** [comment]
superceeded by #128

**Pull request closed without merging**